Automattic/wp-calypso
# Masterbar Site Menu - Select a wp-admin site that the user come from

Related to;
- https://github.com/Automattic/dotcom-forge/issues/8234
- https://github.com/Automattic/wp-calypso/pull/92720/
- https://github.com/Automattic/wp-calypso/pull/92762/

## Proposed Changes

This PR changes to select a site that is specified by the query param `origin_site_id`.
It will be used to select a wp-admin site where the user is coming from. 

## Why are these changes being made?


- See https://github.com/Automattic/dotcom-forge/issues/8234

## Testing Instructions



* Go to `/sites?origin_site_id=<your-site-id>` (you have to use a site id, not a site slug)
* Observe the site menu is present in the masterbar
* Select and Close a site and observe the animation works correctly

To override the previously selected site

* Go to `/sites`
* Select a Default interface site
* Go to My Home of the site
* Click the W icon to come back to `/sites`
* Observe the site menu is present in the masterbar
* Now, reload the window with the query param added: `/sites?origin_site_id=<your-site-id>`
* Observe the site menu is overwritten with the `origin_site_id`

/read and /me*

* Go to `/read?origin_site_id=<your-site-id>` or `/me?origin_site_id=<your-site-id>`
* Observe the site menu is present in the masterbar

## Pre-merge Checklist
